roxie/ccd/ccdcontext.cpp

```diff
@@ -1593,6 +1593,14 @@ class CRoxieContextBase : implements IRoxieAgentContext, implements ICodeContext
                     progressWorkUnit->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTgraph, graphScope, StWhenStarted, NULL, startTimeStamp, 1, 0, StatsMergeAppend);
                     updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StTimeElapsed, graphDesc, elapsedTime);
                     addTimeStamp(progressWorkUnit, SSTgraph, graphName, StWhenFinished, graph->queryWorkflowId());
+
+                    ProcessInfo endProcessInfo(ReadAllInfo);
+                    SystemProcessInfo delta = endProcessInfo - startProcessInfo;
+                    updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StTimeUser, graphDesc, delta.getUserNs());
+                    updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StTimeSystem, graphDesc, delta.getSystemNs());
+                    updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StNumContextSwitches, graphDesc, delta.getNumContextSwitches());
+                    updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StSizeMemory, graphDesc, endProcessInfo.getActiveResidentMemory());
+                    updateWorkunitStat(progressWorkUnit, SSTgraph, graphScope, StSizePeakMemory, graphDesc, endProcessInfo.getPeakResidentMemory());
```

Comment:
in other contexts (eclcc, hqlcpp) this stat is only written if getPeakResidentMemory() is non zero, should this be the same?

Reply:
I think it is very unlikely to be 0 - only if the function failed.  Probably not worth making it conditional.

Reply:
```
        if (processFinishTime.getPeakResidentMemory())
            updateWorkunitStat(instance.wu, SSTcompilestage, scopeName, StSizePeakMemory, NULL, processFinishTime.getPeakResidentMemory());
```

agree, but it's conditional in other places, in eclcc and hqlecl ?